### PR TITLE
Tempus: starting step size for ERK method

### PR DESCRIPTION
**Pull request by sconde**
- add RK starting step size routine
- use new starting stepsize routine only when Embedded/adaptive
timestepping
- In ERK stepper, explicit set the time in workingState metadata.
- update ERK unit test since the refIstep is now lower due to optimal
  starting step size computed

@trilinos/tempus 

## Description

Compute an optimal `Initial Time Step` for ERK. It could also be used for DIRK steppers. Only cost 2 RHS evaluations and a forward euler step.

## Related Issues

* Closes 
* Blocks 
* Is blocked by 
* Follows 
* Precedes 
* Related to 
* Part of 
* Composed of 

## Checklist

- [ ] My commit messages mention the appropriate GitHub issue numbers.
- [ ] My code follows the code style of the affected package(s).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [code contribution guidelines](../blob/master/CONTRIBUTING.md) for this project.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] No new compiler warnings were introduced.
- [ ] These changes break backwards compatibility.

### CODE REVIEW

File: packages/tempus/src/Tempus_StepperRKStartingStepSize.hpp [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+     Thyra::ModelEvaluatorBase::InArgs<Scalar> inArgs_ = inModel->getNominalValues();
+     Thyra::ModelEvaluatorBase::OutArgs<Scalar> outArgs_ = inModel->createOutArgs();
+     inArgs_.set_x(stageX_);
+     if (inArgs_.supports(MEB::IN_ARG_t)) inArgs_.set_t(time);
+     if (inArgs_.supports(MEB::IN_ARG_x_dot)) inArgs_.set_x_dot(Teuchos::null);
+     outArgs_.set_f(stageXDot_[0]); // K1
+     inModel->evalModel(inArgs_,outArgs_);
+
+     Scalar d0 = err_func(stageX_, errorRel, errorAbs, scratchX);
+     Scalar d1 = err_func(stageXDot_[0], errorRel, errorAbs, scratchX);
```
Comment (rppawlo) [inline packages/tempus/src/Tempus_StepperRKStartingStepSize.hpp]: Maybe move the err_func definition down to here where it is first used? Would be easier to read/track the algorithm.

File: packages/tempus/src/Tempus_StepperRKStartingStepSize.hpp [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+     // b) first guess for the step size
+     Scalar dt = Teuchos::as<Scalar>(0.01)*(d0/d1);
+     if ((d0 < 1e-5) or (d1 < 1e-5)) dt = 1e-6;
```
Comment (rppawlo) [inline packages/tempus/src/Tempus_StepperRKStartingStepSize.hpp]: the 1e-5 seems problem dependent. If the scaling is bad do these values still work? Should they be adjustable as user input?

File: packages/tempus/src/Tempus_StepperExplicitRK_impl.hpp [hunk 3]
```diff
@@ -134,8 +135,12 @@ void StepperExplicitRK<Scalar>::setObserver(
...
     // Create default observer, otherwise keep current observer.
     if (stepperExplicitRKObserver_ == Teuchos::null) {
-      stepperExplicitRKObserver_ =
-        Teuchos::rcp(new StepperExplicitRKObserver<Scalar>());
+       if (stepperPL_->get<bool>("Use Embedded",false))
+          stepperExplicitRKObserver_ =
+             Teuchos::rcp(new StepperRKStartingStepSize<Scalar>());
+       else
+          stepperExplicitRKObserver_ =
+             Teuchos::rcp(new StepperExplicitRKObserver<Scalar>());
```
Comment (rppawlo) [inline packages/tempus/src/Tempus_StepperExplicitRK_impl.hpp]: I may be mistaken here but it seems like this optimal step size is only used if the observer is not set and only for embedded algorithm. Should we have more control here to set to use the optimal step size regardless of whether observers are set? Do users have the option to override this if they want to force their own step size instead while using the embedded algorithm?

File: packages/tempus/src/Tempus_StepperRKStartingStepSize.hpp (hunk 2 above)
Comment (sconde) [reply packages/tempus/src/Tempus_StepperRKStartingStepSize.hpp]: I can make them adjustable.

**Comment by sconde** [comment]
> We do not want to set the time in a Stepper, so we should remove the lines

@ccober6 I missed that after debugging. It's removed now.

**Pull request merged**